pkg/providers/imagefamily/ubuntu_2004_test.go

```diff
@@ -0,0 +1,102 @@
+			SubscriptionID:                 "test-subscription",
+			ResourceGroup:                  "test-rg",
+			ClusterResourceGroup:           "test-cluster-rg",
+			KubeletClientTLSBootstrapToken: "test-token",
+			KubernetesVersion:              "1.31.0",
+		},
+	}
+
+	kubeletConfig := &bootstrap.KubeletConfiguration{MaxPods: int32(110)}
+	taints := []v1.Taint{{Key: "test", Value: "value", Effect: v1.TaintEffectNoSchedule}}
+	startupTaints := []v1.Taint{{Key: "startup", Value: "value", Effect: v1.TaintEffectNoSchedule}}
+	labels := map[string]string{"test-label": "test-value"}
+	instanceType := &cloudprovider.InstanceType{
+		Name: "Standard_D2s_v3",
+		Capacity: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("2"),
+			v1.ResourceMemory: resource.MustParse("8Gi"),
+		},
+	}
+	imageDistro := "aks-ubuntu-fips-containerd-20.04-gen2"
+	storageProfile := "ManagedDisks"
+	nodeBootstrappingClient := &fake.NodeBootstrappingAPI{}
+
+	bootstrapper := ubuntu.CustomScriptsNodeBootstrapping(
+		kubeletConfig,
+		taints,
+		startupTaints,
+		labels,
+		instanceType,
+		imageDistro,
+		storageProfile,
+		nodeBootstrappingClient,
+	)
+
+	// Verify the returned bootstrapper is of the correct type
+	provisionBootstrapper, ok := bootstrapper.(customscriptsbootstrap.ProvisionClientBootstrap)
+	assert.True(t, ok, "Expected customscriptsbootstrap.ProvisionClientBootstrap type")
+
+	// Verify all fields are properly set
+	assert.Equal(t, "test-cluster", provisionBootstrapper.ClusterName)
+	assert.Equal(t, kubeletConfig, provisionBootstrapper.KubeletConfig)
+	assert.Equal(t, taints, provisionBootstrapper.Taints)
+	assert.Equal(t, startupTaints, provisionBootstrapper.StartupTaints)
+	assert.Equal(t, labels, provisionBootstrapper.Labels)
+	assert.Equal(t, "/subscriptions/test/resourceGroups/test/providers/Microsoft.Network/virtualNetworks/vnet/subnets/subnet", provisionBootstrapper.SubnetID)
+	assert.Equal(t, karpv1.ArchitectureAmd64, provisionBootstrapper.Arch)
+	assert.Equal(t, "test-subscription", provisionBootstrapper.SubscriptionID)
+	assert.Equal(t, "test-rg", provisionBootstrapper.ResourceGroup)
+	assert.Equal(t, "test-cluster-rg", provisionBootstrapper.ClusterResourceGroup)
+	assert.Equal(t, "test-token", provisionBootstrapper.KubeletClientTLSBootstrapToken)
+	assert.Equal(t, "1.31.0", provisionBootstrapper.KubernetesVersion)
+	assert.Equal(t, imageDistro, provisionBootstrapper.ImageDistro)
+	assert.Equal(t, instanceType, provisionBootstrapper.InstanceType)
+	assert.Equal(t, storageProfile, provisionBootstrapper.StorageProfile)
+	assert.Equal(t, nodeBootstrappingClient, provisionBootstrapper.NodeBootstrappingProvider)
+	assert.Equal(t, customscriptsbootstrap.ImageFamilyOSSKUUbuntu2004, provisionBootstrapper.OSSKU, "ImageFamily field must be set to prevent unsupported image family errors")
+}
+
+func TestUbuntu2004_Name(t *testing.T) {
+	ubuntu := imagefamily.Ubuntu2004{}
+	assert.Equal(t, v1beta1.UbuntuImageFamily, ubuntu.Name())
```

Comment:
I am actually a little surprised by Name returning `v1beta1.UbuntuImageFamily`, but it does seem to line up with the patterning of AzureLinux. 

Curious how `Name()`, is used, and what the value should be for our ImageFamilies.

Reply:
Can you just add a `TODO (charliedmcb): look into .Name() usage, and implications`

To the image family func itself:
<img width="298" height="66" alt="image" src="https://github.com/user-attachments/assets/e58d7470-a9ce-4d30-8bc8-4a1c1a99f3c8" />

Appears to just be leveraged on error metrics